core/src/mindustry/logic/LExecutor.java

```diff
@@ -1987,17 +1987,15 @@ public void run(LExecutor exec){
                 var marker = state.markers.get(exec.numi(id));
                 if(marker == null) return;
 
-                if(type == LMarkerControl.text){
-                    marker.setText((exec.obj(p1) != null ? exec.obj(p1).toString() : "null"), false);
-                }else if(type == LMarkerControl.flushText){
-                    marker.setText(exec.textBuffer.toString(), true);
+                if(type == LMarkerControl.flushText){
+                    marker.setText(exec.textBuffer.toString(), exec.bool(p1));
                     exec.textBuffer.setLength(0);
                 }else if(type == LMarkerControl.texture){
-                    if(exec.obj(p1) != null){
-                        StringBuilder res = new StringBuilder(exec.obj(p1).toString());
-                        if(exec.obj(p2) != null) res.append("-").append(exec.obj(p2).toString());
-                        if(exec.obj(p3) != null) res.append("-").append(exec.obj(p3).toString());
-                        marker.setTexture(res.toString());
+                    if(exec.bool(p1)){
+                        marker.setTexture(exec.textBuffer.toString());
+                        exec.textBuffer.setLength(0);
+                    }else{
+                        marker.setTexture((exec.obj(p2) != null ? exec.obj(p2).toString() : "null"));
```

Comment:
The outputs for toString() are different than what logic outputs when you try to print an object, so this should use `PrintI#toString` instead.